tests/publiccloud/storage_perf.pm

```diff
@@ -0,0 +1,144 @@
+    my $size    = get_var('SSD_SIZE', 100);
+
+    my @scenario = (
+        {
+            name      => 'reference',
+            rw        => 'randread',
+            rwmixread => '100',
+            bs        => '8k'
+        },
+        {
+            name      => 'reallife',
+            rw        => 'randrw',
+            rwmixread => '65',
+            bs        => '8k'
+        },
+        {
+            name      => 'writeintensive',
+            rw        => 'randwrite',
+            rwmixread => '10',
+            bs        => '8k'
+
+        },
+        {
+            name      => 'maxthroughput',
+            rw        => 'read',
+            rwmixread => '100',
+            bs        => '64k'
+        }
+    );
+    $self->select_serial_terminal;
+
+    my $provider = $self->provider_factory();
+    my $instance = $provider->create_instance();
+    $instance->run_ssh_command(cmd => 'sudo SUSEConnect -r ' . $reg_code, timeout => 600);
+    $instance->run_ssh_command(cmd => 'sudo zypper --gpg-auto-import-keys -q in -y fio', timeout => 600);
+
+    my $block_device = '/dev/' . $instance->run_ssh_command(cmd => 'lsblk|grep ' . $size . '|cut -f 1 -d " "');
+    record_info('dev', "Block device under test: $block_device");
+
+    for my $href (@scenario) {
+        print "$href->{name}\n";
+        record_info('FIO', 'Running test case "' . $href->{name} . '"');
+
+        my $cmd = 'sudo fio --name=' . $href->{name};
+        $cmd .= ' --direct=1';
+        $cmd .= ' --ioengine=libaio';
+        $cmd .= ' --rw=' . $href->{rw};
+        $cmd .= ' --rwmixread=' . $href->{rwmixread};
+        $cmd .= ' --bs=' . $href->{bs};
+        $cmd .= ' --runtime=' . $runtime;
+        $cmd .= ' --filename=' . $block_device;
+        $cmd .= ' --iodepth=' . IODEPTH;
+        $cmd .= ' --numjobs=' . NUMJOBS;
+        $cmd .= ' --output-format=json';
+        $cmd .= ' --group_reporting';
+
+        record_info('cmd', $cmd);
+        my $output = $instance->run_ssh_command(cmd => $cmd, timeout => $runtime + 60);
+        record_info('Result', $output);
+
+        # TODO: Parse results and push them to performance DB
+    }
+}
+
+1;
+
+=head1 Discussion
+
+Test module to run Storage Performance using FIO on publiccloud.
+4 Scenarios are defined:
+ - Reference: random reads with 8 KB blocks.
+       See how the disk behaves with only random reads.
+ - Real life: 65% random reads with 8 KB blocks.
+       Approximate to real applications with balance between reads and writes.
+ - Write Intensive: 90% random writes with 8 KB blocks.
+       See how the disk behaves with intestive block writting.
+ - Max Throughput: 100% sequential reads with big block size.
+       Try to reach maximum throughput of the block devices.
+
+Results to be reported for each scenario:
+ - throughput
+ - latency
+ - IOPS
+
+=head1 Configuration
+
+=head2 PUBLIC_CLOUD_FIO
+
+If set, this test module is added to the job.
+
+=head2 PUBLIC_CLOUD_PROVIDER
```

Comment:
I'm not sure if we need to describe the "standard" parameters within each test?
Maybe we could just redirect to `lib::publiccloud::basetest` and describe it once there?
WDYT?

Reply:
yes, I just added the local parameters in this .pm. But we need to do the same in ipa.pm